File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/UserLoginIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;

class UserLoginIdFlattenedDto
{

    const PROPERTIES = [
            'userId' => 'string',
            'loginKeyLoginProvider' => 'string',
            'loginKeyProviderKey' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var UserLoginId
     */
    private $value;

    /**
     * @param UserLoginId $value
     */
    public function __construct(UserLoginId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new UserLoginId();
        }
    }

    /**
     * @return UserLoginId
     */
    public function toUserLoginId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getUserId()
    {
        return $this->value->getUserId();
    }

    /**
     * @param string $userId
     */
    public function setUserId($userId)
    {
        $this->value->setUserId($userId);
    }

    /**
     * @return string
     */
    public function getLoginKeyLoginProvider()
    {
        return $this->value->getLoginKey()->getLoginProvider();
    }

    /**
     * @param string $loginKeyLoginProvider
     */
    public function setLoginKeyLoginProvider($loginKeyLoginProvider)
    {
        $this->value->getLoginKey()->setLoginProvider($loginKeyLoginProvider);
    }

    /**
     * @return string
     */
    public function getLoginKeyProviderKey()
    {
        return $this->value->getLoginKey()->getProviderKey();
    }

    /**
     * @param string $loginKeyProviderKey
     */
    public function setLoginKeyProviderKey($loginKeyProviderKey)
    {
        $this->value->getLoginKey()->setProviderKey($loginKeyProviderKey);
    }


}

File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/UserLoginIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class UserLoginIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "userId",
            "loginKeyLoginProvider",
            "loginKeyProviderKey",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private UserLoginId value;

    public UserLoginIdFlattenedDto()
    {
        this(new UserLoginId());
    }

    public UserLoginIdFlattenedDto(UserLoginId value)
    {
        this.value = value;
    }

    public UserLoginId toUserLoginId()
    {
        return this.value;
    }


    public String getUserId()
    {
        return this.value.getUserId();
    }

    public void setUserId(String userId)
    {
        this.value.setUserId(userId);
    }

    public String getLoginKeyLoginProvider()
    {
        return this.value.getLoginKey().getLoginProvider();
    }

    public void setLoginKeyLoginProvider(String loginKeyLoginProvider)
    {
        this.value.getLoginKey().setLoginProvider(loginKeyLoginProvider);
    }

    public String getLoginKeyProviderKey()
    {
        return this.value.getLoginKey().getProviderKey();
    }

    public void setLoginKeyProviderKey(String loginKeyProviderKey)
    {
        this.value.getLoginKey().setProviderKey(loginKeyProviderKey);
    }


}

